skimage/data/__init__.py

```diff
@@ -49,6 +54,209 @@
            'shepp_logan_phantom',
            'stereo_motorcycle']
 
+legacy_data_dir = osp.abspath(osp.dirname(__file__))
+skimage_distribution_dir = osp.join(legacy_data_dir, '..')
+
+try:
+    import pooch
+    from pooch.utils import file_hash
+    has_pooch = True
+except ImportError:
+    has_pooch = False
+    import hashlib
+
+    def file_hash(fname):
+        # Versioned (i.e. copied from) from pooch.utils
+        # Calculate the hash in chunks to avoid overloading the memory
+        chunksize = 65536
+        hasher = hashlib.sha256()
+        with open(fname, "rb") as fin:
+            buff = fin.read(chunksize)
+            while buff:
+                hasher.update(buff)
+                buff = fin.read(chunksize)
+        return hasher.hexdigest()
+
+
+def _hash_hash(path, expected_hash):
+    """Check if the provided path has the expected hash."""
+    if not osp.exists(path):
+        return False
+    return file_hash(path) == expected_hash
+
+
+def _fetch(data_filename):
```

Comment:
so, don't we want to expose this function at all? The question is whether we want to return arrays for all datasets, or whether in some cases we are just happy downloading the image and letting the user to choose what to do (for large datasets, loading in memory only a chunck of the array etc.). Anyway this can be decided later on and should not block the PR, it's easier to add a function to the API than removing it ;-).

Reply:
If I'm correct, the current behavior (download only) has the advantage to reduce the bandwidth consumption (or I misunderstood your comment).

Reply:
Maybe my comment was not clear :-). I was just wondering if we should expose this `_fetch` function in the public API, since it allows to download one specific file. At the moment we have access to the `download_all` function, and to `load` which calls `imread` on the downloaded file. For specific file formats not compatible with `imread` this might not work. But it's ok, this can be added later.